projects/packages/forms/src/dashboard/inbox/response.js

```diff
@@ -19,39 +20,166 @@ const isFileUploadField = value => {
-									<Button variant="link" href={ file.url } target="_blank">
-										{ decodeEntities( file.name ) } | { file.size }
-									</Button>
-								</div>
-							);
-					  } )
-					: '-' }
+const PreviewFile = ( { file, isLoading, onImageLoaded } ) => {
+	const imageClass = clsx( 'jp-forms__inbox-file-preview-container', {
+		'is-loading': isLoading,
+	} );
+
+	return (
+		<div className="jp-forms__inbox-file-preview-shell">
+			{ isLoading && (
+				<div className="jp-forms__inbox-file-loading">
+					<Spinner className="jp-forms__inbox-spinner" />
+					<div className="jp-forms__inbox-file-loading-message ">
+						{ __( 'Loading preview…', 'jetpack-forms' ) }
+					</div>
+				</div>
+			) }
+
+			<div className={ imageClass }>
+				<img
+					src={ file.url }
+					alt={ decodeEntities( file.name ) }
+					onLoad={ onImageLoaded }
+					className="jp-forms__inbox-file-preview-image"
+				/>
 			</div>
-		);
-	}
+		</div>
+	);
+};
 
-	// Emails
-	const emailRegex = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
-	if ( emailRegex.test( value ) ) {
-		return <a href={ `mailto:${ value }` }>{ value }</a>;
-	}
+const FileField = ( { file, onClick, key } ) => {
+	const fileExtension = file.name.split( '.' ).pop().toLowerCase();
+	const fileType = file.type.split( '/' )[ 0 ];
+
+	const iconMap = {
+		image: 'png',
+		video: 'mp4',
+		audio: 'mp3',
+		document: 'pdf',
+		application: 'txt',
+	};
 
-	return value;
+	const extensionMap = {
+		pdf: 'pdf',
+		png: 'png',
+		jpg: 'png',
+		jpeg: 'png',
+		gif: 'png',
+		mp4: 'mp4',
+		mp3: 'mp3',
+		webm: 'webm',
+		doc: 'doc',
+		docx: 'doc',
+		txt: 'txt',
+		ppt: 'ppt',
+		pptx: 'ppt',
+		xls: 'xls',
+		xlsx: 'xls',
+		csv: 'xls',
+		zip: 'zip',
+		sql: 'sql',
+		cal: 'cal',
+	};
+	const iconType = extensionMap[ fileExtension ] || iconMap[ fileType ] || 'txt';
+	const iconClass = clsx( 'file-field__icon', 'icon-' + iconType );
+	return (
+		<div key={ key } className="file-field__item">
+			<div className="file-field__info">
+				<div className={ iconClass }></div>
+				<div className="file-field__name">
+					{ file.is_previewable && (
+						<Button target="_blank" variant="link" onClick={ onClick }>
+							{ decodeEntities( file.name ) }
+						</Button>
+					) }
+					{ ! file.is_previewable && (
+						<ExternalLink href={ file.url + '&preview=true' }>
+							{ decodeEntities( file.name ) }
+						</ExternalLink>
+					) }
+					<div className="file-field__meta-info">
+						{ sprintf(
+							/* translators: %1$s size of the file and %2$s is the file extension */
+							__( '%1$s, %2$s', 'jetpack-forms' ),
+							file.size,
+							fileExtension.toUpperCase()
+						) }
+					</div>
+				</div>
+			</div>
+			<span className="file-field__item-actions">
+				<Tooltip text={ __( 'Download', 'jetpack-forms' ) }>
+					<Button variant="secondary" href={ file.url } target="_blank">
+						<Icon icon={ download } />
+					</Button>
+				</Tooltip>
+			</span>
+		</div>
+	);
 };
```

Comment:
PreviewFile and FileField are decent sized components in their own right. My preference might be to put them in their own component files rather than have multiple components in a one file. Could be directly in this folder, or we could create a inbox/dataviews/components folder. 

Not a requirement, and if you feel otherwise, fine to roll as is.

Reply:
Good suggestion. I will do this in a follow up.